java-frontend/src/main/java/org/sonar/java/AnalyzerMessage.java

```diff
@@ -118,12 +132,14 @@ public String toString() {
   }
 
   private static AnalyzerMessage.TextSpan textSpanBetween(SyntaxToken firstSyntaxToken, SyntaxToken lastSyntaxToken) {
-    return new AnalyzerMessage.TextSpan(
+    AnalyzerMessage.TextSpan location = new AnalyzerMessage.TextSpan(
       firstSyntaxToken.line(),
       firstSyntaxToken.column(),
       lastSyntaxToken.line(),
       lastSyntaxToken.column() + lastSyntaxToken.text().length()
     );
+    Preconditions.checkState(!location.isEmpty(), "Issue location should not be empty");
```

Comment:
I tend to think that the message should be a bit more helpful than that... if it blows, message is not really helpful.